Challenge1/src/agent.spec.ts

```diff
@@ -1,74 +1,127 @@
-// import {
-//   FindingType,
-//   FindingSeverity,
-//   Finding,
-//   HandleTransaction,
-//   createTransactionEvent,
-//   ethers,
-// } from "forta-agent";
-// import agent, {
-//   ERC20_TRANSFER_EVENT,
-//   TETHER_ADDRESS,
-//   TETHER_DECIMALS,
-// } from "./agent";
+import { Finding, HandleTransaction, FindingSeverity, FindingType, TransactionEvent, ethers } from "forta-agent";
+import { createAddress } from "forta-agent-tools";
+import { TestTransactionEvent } from "forta-agent-tools/lib/test";
+import { provideHandleTransaction } from "./agent";
+import { CREATE_AGENT_ABI, UPDATE_AGENT_ABI, OTHER_FUNCTION_ABI, CHAIN_IDS } from "./constants";
 
-// describe("high tether transfer agent", () => {
-//   let handleTransaction: HandleTransaction;
-//   const mockTxEvent = createTransactionEvent({} as any);
+const mockNethermindAddress = createAddress("0x01");
+const mockFortaRegistryAddress = createAddress("0x02");
+const mockOtherAddress = createAddress("0x03");
 
-//   beforeAll(() => {
-//     handleTransaction = agent.handleTransaction;
-//   });
+const MOCK_AGENT_ID = 1;
 
-//   describe("handleTransaction", () => {
-//     it("returns empty findings if there are no Tether transfers", async () => {
-//       mockTxEvent.filterLog = jest.fn().mockReturnValue([]);
+const ifaceForSetData = new ethers.utils.Interface([CREATE_AGENT_ABI, UPDATE_AGENT_ABI, OTHER_FUNCTION_ABI]);
+const encodedCreateAgentData = ifaceForSetData.encodeFunctionData("createAgent", [
+  MOCK_AGENT_ID,
+  mockNethermindAddress,
+  "metadata",
+  CHAIN_IDS,
+]);
+const encodedUpdateAgentData = ifaceForSetData.encodeFunctionData("updateAgent", [
+  MOCK_AGENT_ID,
+  "metadata",
+  CHAIN_IDS,
+]);
+const encodedOtherFunctionData = ifaceForSetData.encodeFunctionData("isRegistered", [MOCK_AGENT_ID]);
 
-//       const findings = await handleTransaction(mockTxEvent);
+// Second transaction for tests 6 and 7, to test if bot picks up only the relevant function
+const mockTxEvent2 = new TestTransactionEvent();
+mockTxEvent2.setFrom(mockNethermindAddress).setTo(mockFortaRegistryAddress).setData(encodedOtherFunctionData);
 
-//       expect(findings).toStrictEqual([]);
-//       expect(mockTxEvent.filterLog).toHaveBeenCalledTimes(1);
-//       expect(mockTxEvent.filterLog).toHaveBeenCalledWith(
-//         ERC20_TRANSFER_EVENT,
-//         TETHER_ADDRESS
-//       );
-//     });
+describe("Nethermind Bot Creation and Update Detection Bot Test Suite", () => {
+  let handleTransaction: HandleTransaction;
+  let mockTxEvent = new TestTransactionEvent();
+  beforeAll(() => {
+    handleTransaction = provideHandleTransaction(
+      CREATE_AGENT_ABI,
+      UPDATE_AGENT_ABI,
```

Comment:
Would also request that a pair of mock ABI variables also be declared in this test file.